글 작성 및 조회

beetrbgus opened the pull request:
- 필요 Entity, Dto, Enum 추가 
- Controller, Service, Repository 추가 
- Mapstruct Mapper 추가

beetrbgus committed:
Skill refactoring

- 변수명 변경 ( Front에서 가독성 좋게 변경 요청)
- SkillCategory 생성
- User의 profile 버킷 키 때문에 오류 나는 부분 수정. 및 url 경로 단순화

beetrbgus committed:
Post에서 필요한 Enum 객체 생성

- PostStatus : 모집글의 게시,삭제 등의 상태
- PostType : 해당 모집글의 모집분류
- ProgressType : 진행 방법

beetrbgus committed:
모집글 작성시 필요 객체 생성

- 모집글 작성시 필요한 모집글의 기술스택 객체와 모집 분야 객체
- Create 시 Dto
- Dto 를 Entity 로 변환해주는 Mapper

beetrbgus committed:
모집글 작성시 필요 Repository 생성

- save 시에는 Spring Data JPA 사용하기 위해 JPARepository 상속하는 interface 생성

beetrbgus committed:
모집글 생성 및 조회

- Controller 및 Service 작성
- 조회시 객체 탐색 대신 Fetch Join을 사용하기 위한 커스텀 레파지토리 생성
- 반환 객체 생성
- 에러 코드 생성 및 예외클래스 생성

jjoylee commented on src/main/java/com/justcodeit/moyeo/study/persistence/Recruitment.java:
```diff
@@ -0,0 +1,47 @@
+    private Post post;
+
+    public Recruitment(Long id, RecruitType recruitType, int recruitPeopleNum) {
+        this.id = id;
```
id는 autogenerate인데 생성자로 받아서 사용할 필요가 있는지 궁금합니다.

beetrbgus replied:
Mapstruct 가 데이터 옮기는 방식이 생성자, 빌더, Setter 이렇게 있는데, 지금 여기서는 생성자로 써서 생성자에 추가했습니다.

jjoylee commented on src/main/java/com/justcodeit/moyeo/study/application/post/PostService.java:
```diff
@@ -0,0 +1,66 @@
+    private final PostCustomRepository postCustomRepository;
+    private final RecruitmentRepository recruitmentRepository;
+
+    @Transactional(readOnly = true)
```
insert 작업이라 readonly = true는 의미상 맞지 않다고 생각됩니다.

beetrbgus replied:
체크 - 원래 하려고 생각했던거는 isolation 이었는데 잘못 적음. 근데 isolation도 굳이...? 써야될까 싶어서 그냥 삭제해서 올려요

jjoylee commented on src/main/java/com/justcodeit/moyeo/study/application/post/PostService.java:
```diff
@@ -0,0 +1,66 @@
+        Post post = postMapper.createReqDtoToEntity(postCreateReqDto);
+        postRepository.save(post);
+
+        postCreateReqDto.getSkillIdList()
```
이 부분 로직들은 나눠서 PostSkillService와 SkillService에서 처리하게 해도 좋을 것 같습니다.

beetrbgus replied:
SkillService에서 호출하게요??
그게 저도 맞다고 생각하는데 전회사 팀장님이 Service가 Service를 호출하게 되면 순환참조 문제가 발생할 수도 있고 등등 등의 이유가 있다--라고 해서 분리 안 한거긴 한데,  이 부분은 내일 얘기 해보시는게 어떨까용

jjoylee commented on src/main/java/com/justcodeit/moyeo/study/persistence/repository/querydsl/PostCustomRepositoryImpl.java:
```diff
@@ -0,0 +1,60 @@
+        }
+        return QPost.post.id.lt(postId);
+    }
+    private BooleanExpression defaultShow() {
```
이 메서드는 다른 곳에서 사용하는 부분이 없다면 지워도 괜찮을 것 같습니다. 사용하지 않는 것 같아서요.

beetrbgus replied:
아 까먹었네용..만든다는거 지금 당장!은 안쓰지만 다음 PR에 추가할 예정이에요. 
모집글 우리가 Soft Delete로 하기로 해서 기본적으로 모집글은 '정상 상태' 에서만 보여야해서 메서드로 상태 조회 Expression 만들 계획입니다.

beetrbgus committed:
Review Comment 수정

- 의미없는 readOnly 삭제

beetrbgus committed:
Merge remote-tracking branch 'origin/develop' into feature/221012_post

beetrbgus committed:
Post 생성 Refactoring

- 조회수 생성 추가 및 연관관계는 setter 생성하고 그 안에서 역할에 맞게 로직 수정
-
- Mapstruct Mapper는 예를 들어 PostMapper면 PostMapper.POST_MAPPER.메소드 이름 이면 이상해서 좀 의미가 명확하게
PostMapper.INSTANCE로 변경

beetrbgus committed:
PostService 글 작성 Refactoring

- Post 작성할 때 필요한 역할들을 Post에 위임

beetrbgus committed:
Post 전체 조회 API 필요 Dto 추가

beetrbgus committed:
Post 전체 조회 API 수정

- 작성자의 id 추가되게 수정
- 반환 Mapper 추가
- Join 조건 명시
-
